source/simulator/melt.cc

```diff
@@ -60,6 +60,29 @@ namespace aspect
 
   namespace Assemblers
   {
+    namespace
+    {
+      template<int dim>
+      bool
+      is_stokes_component (const Introspection<dim> &introspection,
```

Comment:
I wonder if that's the right name for this function because it returns `true` for both kinds of pressures. I don't recognize this as the "Stokes problem" then. Maybe call it `is_velocity_or_pressures()`?